hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go

```diff
@@ -640,6 +640,17 @@ func (r *DedicatedServingComponentSchedulerAndSizer) updateHostedCluster(ctx con
 	if sizeConfig.Effects != nil && sizeConfig.Effects.APICriticalPriorityClassName != nil {
 		hc.Annotations[hyperv1.APICriticalPriorityClass] = *sizeConfig.Effects.APICriticalPriorityClassName
 	}
+	if sizeConfig.Effects != nil && sizeConfig.Effects.MachineHealthCheckTimeout != nil {
+		hc.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation] = sizeConfig.Effects.MachineHealthCheckTimeout.Duration.String()
+	} else {
+		// If mhc timeout is configured for any size in the config, remove the annotation
+		// to fallback to the default
+		if configHasMHCTimeout(config) {
+			if hc.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation] != "" {
```

Comment:
do we need the if here, will delete(hc.Annotations, hyperv1.MachineHealthCheckTimeoutAnnotation) panic without it?

Reply:
delete will not panic, but the if as mentioned earlier is to determine if we should remove the annotation

Reply:
I mean the second if `if hc.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation] != "" {`
seems redundant. We could just

		if configHasMHCTimeout(config) {
				delete(hc.Annotations, hyperv1.MachineHealthCheckTimeoutAnnotation)
			}

Reply:
ahh got it
thx!